File: frontend/src/services/leadService.js
import api from './api'

export default {
  // POST /api/v1/leads — ADMIN : { clientId, annonceId, visiteId?, noteAdmin? }
  create(data) {
    return api.post('/leads', data)
  },

  // GET /api/v1/leads?page&size&statut
  getAll(page = 0, size = 20, statut = null) {
    const params = { page, size }
    if (statut) params.statut = statut
    return api.get('/leads', { params })
  },

  // GET /api/v1/leads/{id}
  getById(id) {
    return api.get(`/leads/${id}`)
  },

  // PUT /api/v1/leads/{id}/status — { statut, noteAdmin? }
  updateStatut(id, statut, noteAdmin = null) {
    return api.put(`/leads/${id}/status`, { statut, noteAdmin })
  },
}

File: backend/src/main/java/sn/immosn/backend/lead/service/LeadService.java
package sn.immosn.backend.lead.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import sn.immosn.backend.client.web.lead.dto.*;
import sn.immosn.backend.lead.data.entity.StatutLead;

public interface LeadService {
    LeadResponseDto create(LeadCreateRequestDto request);
    Page<LeadResponseDto> getAll(StatutLead statut, Pageable pageable);
    LeadResponseDto getById(Long id);
    LeadResponseDto updateStatut(Long id, UpdateStatutLeadDto dto);
}
